src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3720,6 +3720,10 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
 
                     return escape;
 
+                case BoundKind.DynamicObjectCreationExpression:
+                    // always returnable
```

Comment:
>// always returnable [](http://example.com/codeflow?start=20&length=20)

Have we actually confirmed that?  For example, Is there anything interesting to check in the initializer? Why are we checking it for a regular object creation?